### PR TITLE
Use proper fs save vertx implementation

### LINKED ISSUE #39: Update Vert.x version to 3.9.0 once it released
The delay can be removed after the update: 

https://github.com/artipie/asto/blob/537c03ff3f11244fa4ef2d12a4657db635fab531/src/main/java/com/artipie/asto/fs/RxFile.java#L108

A ticket with a bug fix: 
https://github.com/vert-x3/vertx-rx/issues/218

### PR DESCRIPTION
**Pull request by Sammers21**
Since vert.x 3.9.0 has been released, now we can use proper RxFile#save implementation

Closes #39

### CODE REVIEW

**Comment by g4s8** [comment]
@Sammers21 #151 was merged

**Comment by Sammers21** [comment]
@g4s8 , ready for review

**Comment by 0crat** [comment]
This pull request [#152](https://github.com/artipie/asto/issues/152) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/78a40f0f-a85c-41c3-91d3-ef704a3e754d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

File: src/main/java/com/artipie/asto/fs/RxFile.java [hunk 1]
```diff
@@ -108,24 +108,26 @@ public RxFile(final Path file, final FileSystem fls) {
...
+                .setRead(false)
+                .setCreate(true)
+                .setWrite(true)
+                .setTruncateExisting(true)
+        )
+            .flatMapCompletable(
+                asyncFile -> Completable.create(
+                    emitter -> {
+                        flow.map(buf -> Buffer.buffer(new Remaining(buf).bytes()))
+                            .subscribe(asyncFile.toSubscriber()
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/fs/RxFile.java:123]: @Sammers21 please move this statement to the next line

File: src/main/java/com/artipie/asto/fs/RxFile.java [hunk 2]
```diff
@@ -108,24 +108,26 @@ public RxFile(final Path file, final FileSystem fls) {
...
+                                .onWriteStreamEnd(emitter::onComplete)
+                                .onWriteStreamError(emitter::onError)
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/fs/RxFile.java:125]: @Sammers21 what's the difference between `onWriteStreamError` and `onError`? Don't we need to wrap stream errors somehow to distinguish from just errors? E.g.:
```java
onWriteStreamError(err -> emitter.onError(new IOException("AsyncFile stream error", err)))
```

**Comment by 0crat** [comment]
This pull request [#152](https://github.com/artipie/asto/issues/152) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/51ed7109-41aa-428c-8367-5d81a2114151); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Sammers21** [comment]
I've found an issue with Vert.x, I will make this PR ready review when the problem will be fixed

**Comment by g4s8** [comment]
@Sammers21 I'm closing this PR, since it's not relevant.

**Comment by 0crat** [comment]
Job `gh:artipie/asto#152` is not assigned, can't get performer

**Comment by Sammers21** [comment]
@g4s8, this PR is relevant. It improves performance by 40%.

**Comment by 0crat** [comment]
Job [#152](https://github.com/artipie/asto/issues/152) is already in scope

**Comment by 0crat** [comment]
GitHub issue is still open, won't close

**Comment by 0crat** [comment]
Code review was too long (38 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Pull request [#152](https://github.com/artipie/asto/issues/152) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**